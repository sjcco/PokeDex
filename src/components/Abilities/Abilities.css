.container {
  height: 100%;
  display: flex;
  flex-direction: column;
  border-radius: 0.5rem;
  box-shadow: 0.05rem 0.1rem 0.3rem -0.03rem rgba(0, 0, 0, 0.45);
  margin: 0.5rem 0.5rem;
  background-color: white;
}

.ability-bar {
  width: 90%;
  border: 1px solid rgb(197, 197, 197);
  border-radius: 0.25rem;
  margin: 1rem auto;
  padding: 0.25rem 0;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-weight: 100;
  font-size: 0.9rem;
  text-align: center;
}

.h-ability-bar {
  width: 90%;
  display: flex;
  border: 1px solid rgb(197, 197, 197);
  border-radius: 0.25rem;
  margin: 0 auto 1rem auto;
  padding: 0;
}

.hidden-title {
  width: 30%;
  margin: 0;
  padding: 0;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-weight: 100;
  font-size: 0.9rem;
  text-align: center;
  line-height: 27px;
  border-right: 1px solid rgb(197, 197, 197);
}

.hidden-bar {
  width: 70%;
  margin: 0;
  padding: 0;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-weight: 100;
  font-size: 0.9rem;
  text-align: center;
  line-height: 27px;
}
