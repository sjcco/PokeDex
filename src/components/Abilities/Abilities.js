import React from 'react';
import PropTypes from 'prop-types';
import {
  container, abilityBar, hAbilityBar, hiddenTitle, hiddenBar,
} from './Abilities.css';

const Abilities = ({ ability, hiddenAbility }) => (
  <div className={container}>
    <h3 className={abilityBar}>{ability}</h3>
    <div className={hAbilityBar}>
      <h3 className={hiddenTitle}>Hidden</h3>
      <h3 className={hiddenBar}>{hiddenAbility}</h3>
    </div>
  </div>
);

Abilities.propTypes = {
  ability: PropTypes.string.isRequired,
  hiddenAbility: PropTypes.string,
};

Abilities.defaultProps = {
  hiddenAbility: 'none',
};

export default Abilities;
