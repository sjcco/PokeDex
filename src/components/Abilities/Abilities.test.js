/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Abilities from './Abilities';

test('renders content', () => {
  const component = renderer.create(<Abilities ability="it" hiddenAbility="workds" />).toJSON();
  expect(component).toMatchSnapshot();
});
