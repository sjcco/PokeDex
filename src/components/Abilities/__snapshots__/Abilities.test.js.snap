// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
<div
  className="container"
>
  <h3
    className="abilityBar"
  >
    it
  </h3>
  <div
    className="hAbilityBar"
  >
    <h3
      className="hiddenTitle"
    >
      Hidden
    </h3>
    <h3
      className="hiddenBar"
    >
      workds
    </h3>
  </div>
</div>
`;
